gossip-itest/src/test/java/org/apache/gossip/SharedDataReplicationControlTest.java

```diff
@@ -0,0 +1,114 @@
+/*
```

Comment:
More unit tests less integreation tests :) GEnerally if you need an end-to-end test with 100+ lines of code to tests 20 lines of codes you are better off with smaller tests.